lib/numorstring/numorstring_test.go

```diff
@@ -84,8 +84,8 @@ func init() {
 		// Protocol tests.  Invalid integer values will be stored as strings.
 		Entry("should accept 0 protocol as int", "0", protocolType, numorstring.ProtocolFromInt(0)),
 		Entry("should accept 255 protocol as int", "255", protocolType, numorstring.ProtocolFromInt(255)),
-		Entry("should accept tcp protocol as string", "\"tcp\"", protocolType, numorstring.ProtocolFromString("tcp")),
-		Entry("should accept tcp protocol as string", "\"tcp\"", protocolType, numorstring.ProtocolFromString("tcp")),
+		Entry("should accept tcp protocol as string", "\"TCP\"", protocolType, numorstring.ProtocolFromString("TCP")),
```

Comment:
Is lower case still allowed?  Should have UTs either way.

My preference would be to be permissive but we should probably sync with whatever k8s does.